lib/chef-dk/command/generator_commands.rb

```diff
@@ -69,6 +69,12 @@ module SharedGeneratorOptions
         :description => "Use to set arbitrary attribute KEY to VALUE in the code_generator cookbook",
         :default => [],
         :proc => Proc.new { |s| generator_args <<  s.split("=").map(&:strip) }
+
+      option :branch,
```

Comment:
@Sgtpluck ok, right, because generating a cookbook now gives you a build cookbook no matter what. So that implies that we need this for `chef generate cookbook` and `chef generate build-cookbook`. But we still don't need it for file/template/resource/etc. So I think the best course of action is to add the option code to both cookbook and generator cookbook, and that should make everything work. I know it means code duplication but I think that's acceptable compared to the alternative of having useless options show up in the help output for the other generators.